Dockerfile.ubuntu

```diff
@@ -109,21 +110,19 @@ RUN apt-get update && apt-get install --no-install-recommends -y \
             nlohmann-json3-dev \
             python2 \
             python2-dev \
-            python-setuptools \
-            python3 \
-            python3-pip \
-            python3-dev \
-            python-is-python3 \
-            python3-setuptools \
-            python3-virtualenv \
-            python3-numpy \
-            python-is-python3 \
             unzip \
             vim \
             xz-utils && \
             apt-get clean && \
             rm -rf /var/lib/apt/lists/*
 
+RUN apt-get update && apt-get install -y software-properties-common gpg gpg-agent --no-install-recommends && \
+    add-apt-repository ppa:deadsnakes/ppa && \
+    apt-get install -y python3.8 python3.8-dev python3.8-distutils python3-pip --no-install-recommends && \
+    apt-get clean && rm -rf /var/lib/apt/lists/* && \
+    update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.8 1 && \
+    python3 -m pip install numpy==1.21.0 --no-cache-dir
```

Comment:
Why do we need numpy now?

Reply:
It is a requirement for TFS component building in bazel